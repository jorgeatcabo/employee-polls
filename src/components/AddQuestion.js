import { useState } from "react";
import { connect } from "react-redux";
//import { handleAddTweet } from "../actions/tweets";
//import { useNavigate } from "react-router-dom";

const AddQuestion = ({ dispatch, id }) => {
  //const navigate = useNavigate();
  const [text, setText] = useState("");

  const [firstOption, setFirstOption]=useState("")
  const [secondOption, setSecondOption]=useState("")

  const handleFirstOption = (e) => {        
    setFirstOption( e.target.value);
    };

    const handleSecondOption = (e) => {        
      setSecondOption( e.target.value);
    };
    const handleSubmitClick=(e)=>{
      e.preventDefault();
    }


  const handleChange = (e) => {
    const text = e.target.value;

    setText(text);
  };

  const handleSubmit = (e) => {
    e.preventDefault();

   // dispatch(handleAddTweet(text, id));

    setText("");

    if (!id) {
      //navigate("/");
    }
  };


  return (
    <div className='d-flex justify-content-center'>
      
        <div className="card col-12 col-lg-4 login-card mt-2 hv-center">
        <h2>Would You Rather</h2>
        <h6>Create Your Own Poll</h6>
            <form>
                <div className="form-group text-left">
                <label htmlFor="first-option">First Option</label>
                <input type="text" 
                    className="form-control" 
                    id="first-option"             
                    placeholder="Enter First Option" 
                    value={firstOption}
                    onChange={handleFirstOption}
                />
                </div>
                <div className="form-group text-left">
                <label htmlFor="second-option">Second Option</label>
                <input type="text" 
                    className="form-control" 
                    id="second-option" 
                    placeholder="Enter Second Option"
                    value={secondOption}
                    onChange={handleSecondOption} 
                />
                </div>
                <div className="form-check">
                </div>
                <button 
                    type="submit" 
                    className="btn btn-primary"
                    onClick={handleSubmitClick}
                >Submit</button>
            </form>
        {/* <div className="alert alert-success mt-2" style={{display: state.successMessage ? 'block' : 'none' }} role="alert">
            {state.successMessage}
        </div> */}
        </div>
    </div>
  );
};

export default connect()(AddQuestion);
