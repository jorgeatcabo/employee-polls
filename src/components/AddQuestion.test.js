import {render} from '@testing-library/react'
import reducer from "../reducers";
import * as React from 'react';
import { createStore } from "redux";
import { Provider } from "react-redux";
import AddQuestion from './AddQuestion'
import { BrowserRouter as Router } from "react-router-dom";

const store = createStore(reducer);

describe('AddQuestion',()=>{
    it('will match', ()=>{
        var view =render(
            <Provider store={store}>
                <Router>
                    <AddQuestion />
                </Router>
            </Provider>
        );
        expect(view).toMatchSnapshot();
    });
  
})
