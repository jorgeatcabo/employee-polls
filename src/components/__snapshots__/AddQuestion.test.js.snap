// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddQuestion will match 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="d-flex justify-content-center"
      >
        <div
          class="card col-12 col-lg-4 login-card mt-2 hv-center"
        >
          <h2>
            Would You Rather
          </h2>
          <h6>
            Create Your Own Poll
          </h6>
          <form>
            <div
              class="form-group text-left"
            >
              <label
                for="first-option"
              >
                First Option
              </label>
              <input
                class="form-control"
                id="first-option"
                placeholder="Enter First Option"
                required=""
                type="text"
                value=""
              />
            </div>
            <div
              class="form-group text-left"
            >
              <label
                for="second-option"
              >
                Second Option
              </label>
              <input
                class="form-control"
                id="second-option"
                placeholder="Enter Second Option"
                required=""
                type="text"
                value=""
              />
            </div>
            <div
              class="form-check"
            />
            <button
              class="btn btn-primary"
              type="submit"
            >
              Submit
            </button>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="d-flex justify-content-center"
    >
      <div
        class="card col-12 col-lg-4 login-card mt-2 hv-center"
      >
        <h2>
          Would You Rather
        </h2>
        <h6>
          Create Your Own Poll
        </h6>
        <form>
          <div
            class="form-group text-left"
          >
            <label
              for="first-option"
            >
              First Option
            </label>
            <input
              class="form-control"
              id="first-option"
              placeholder="Enter First Option"
              required=""
              type="text"
              value=""
            />
          </div>
          <div
            class="form-group text-left"
          >
            <label
              for="second-option"
            >
              Second Option
            </label>
            <input
              class="form-control"
              id="second-option"
              placeholder="Enter Second Option"
              required=""
              type="text"
              value=""
            />
          </div>
          <div
            class="form-check"
          />
          <button
            class="btn btn-primary"
            type="submit"
          >
            Submit
          </button>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
